### PR TITLE
Devise Authentication

### PR DESCRIPTION
**Pull request by Teshager21**
- Implemented authentication using Devise.
- implemented helper methods to conditionally display links.

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<h1 class='title'>Posts</h1>
+<ul class='container'>
+  <% @posts.each do |post| %>
+    <li class="card m-3 lead p-3"><%= post.body %>
+    <%if post.user && signed_in?%>
```
Comment (maelfosso) [inline app/views/posts/index.html.erb]: use a view helper for that.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -10,6 +10,40 @@
...
+      <ul class="navbar-nav ml-auto">
+        <%if !signed_in?%>
+        <%= link_to  "Login", new_user_session_path ,class:"nav-item nav-link"%>
+        <%= link_to  "Signup ", new_user_registration_path ,class:"nav-item nav-link"%>
+  
+     <%else%>
+
+    <%= link_to  "Logout", destroy_user_session_path , method: :delete, data: {confimation: "are you sure?"}, class:"nav-item" %>
+      
+      <%end%>
```
Comment (maelfosso) [inline app/views/layouts/application.html.erb]: Use a view helper for that `if/else` block

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Nice job :clap: 
However, There some missing requirements.

1. According to [that](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_prs.png), your PR must have a title and a description

2. Read the comment left below

Kindly make the changes and submit a new code review request
Thanks

Mael FOSSO

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<h1 class='title'>Posts</h1>
+<ul class='container'>
+  <% @posts.each do |post| %>
+    <li class="card m-3 lead p-3"><%= post.body %>
+    <!--<%if post.user && signed_in?%>
+      <p>created by <%= post.user.name %></p>
+    <%end %>  -->
```
Comment (uche-inyama) [inline app/views/posts/index.html.erb]: Remove commented code.

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes required

Hi team :wave: 

Kindly remove the commented code between lines 5 and 7 of posts/index.html.erb.

When you are done with this change, submit another


#### ``` Have a great day```

``` Uchechukwu Inyama Henry ``` form for a code.

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

Thank you for applying the changes 
You can merge it :+1: 

Good continuation :handshake: 

Mael FOSSO

**Pull request merged**